Source/Core/AudioCommon/AlsaSoundStream.cpp

```diff
@@ -48,29 +48,134 @@ void AlsaSound::SoundLoop()
   Common::SetCurrentThreadName("Audio thread - alsa");
   while (m_thread_status.load() != ALSAThreadStatus::STOPPING)
   {
+    int err;
+    const snd_pcm_channel_area_t* areas;
+    snd_pcm_uframes_t frames;
+    snd_pcm_uframes_t offset;
+    snd_pcm_uframes_t avail;
+    snd_pcm_uframes_t size;
+    snd_pcm_state_t state;
+
     while (m_thread_status.load() == ALSAThreadStatus::RUNNING)
     {
-      m_mixer->Mix(mix_buffer, frames_to_deliver);
-      int rc = snd_pcm_writei(handle, mix_buffer, frames_to_deliver);
-      if (rc == -EPIPE)
+      state = snd_pcm_state(m_handle);
```

Comment:
Between this state and the other functionality (mmap?) the function has gotten rather large.  Please consider splitting it out into a couple smaller functions to make this `SoundLoop` function more readable